.sandcastle/Dockerfile

```
@@ -15,21 +19,76 @@ RUN curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg \
+# container with --user $hostUid:$hostGid (e.g. 501:20), which does NOT match
+# this UID; the chmod near the end of this file makes /home/agent writable for
+# whatever UID the container actually runs as.
+RUN useradd --create-home --uid 1000 --shell /bin/bash agent
+
 USER agent
+WORKDIR /home/agent
 
-# Install Claude Code CLI
-RUN curl -fsSL https://claude.ai/install.sh | bash
+# Install mise (single-user install under /home/agent/.local).
+RUN curl https://mise.run | sh
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)